idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+
+;=====CH centroid in lat/lon=======
+							centlat=lat[cent[0],cent[1]]
+							centlon=lon[cent[0],cent[1]]
+
+;====caluclate the mean magnetic field=====
+							mB=mean(hd[pos[*,0],pos[*,1]])
+							if mB ge 0 then smb=4 else smb=5
+							mBpos=total(npix[where(magpol gt 0)]*magpol[where(magpol gt 0)])/total(npix[where(magpol gt 0)])
+							mBneg=total(npix[where(magpol lt 0)]*magpol[where(magpol lt 0)])/total(npix[where(magpol lt 0)])
+
+;====insertions of CH properties into property array=====
+							props[0,ident+1]=strcompress(ident,/remove_all)
+							props[1,ident+1]=string(strcompress(arccent0,/remove_all),format='(I10.0)')
+							props[2,ident+1]=string(strcompress(arccent1,/remove_all),format='(I10.0)')
+							props[3,ident+1]=string(strcompress(Xeb,/remove_all),format='(I10.0)')
+							props[4,ident+1]=string(strcompress(Yeb,/remove_all),format='(I10.0)')
+							props[5,ident+1]=string(strcompress(Xwb,/remove_all),format='(I10.0)')
+							props[6,ident+1]=string(strcompress(Ywb,/remove_all),format='(I10.0)')					
+							props[7,ident+1]=string(strcompress(Xnb,/remove_all),format='(I10.0)')
+							props[8,ident+1]=string(strcompress(Ynb,/remove_all),format='(I10.0)')
+							props[9,ident+1]=string(strcompress(Xsb,/remove_all),format='(I10.0)')
+							props[10,ident+1]=string(strcompress(Ysb,/remove_all),format='(I10.0)')
+							props[11,ident+1]=string(strcompress(width,/remove_all),format='(I10.0)')
+							armm=string(strcompress((trummar/(1e+12)),/remove_all),format='(e11.1)')
+							props[12,ident+1]=strmid(armm,0,8)+strmid(armm,9,2)
+							props[13,ident+1]=string(strcompress((arcar*100/(!PI*rs^(2))),/remove_all),format='(F10.1)')
+							props[14,ident+1]=string(strcompress(mB,/remove_all),format='(F10.1)')
+							props[15,ident+1]=string(strcompress(mBpos,/remove_all),format='(F10.1)')
+							props[16,ident+1]=string(strcompress(mBneg,/remove_all),format='(F10.1)')
+							props[17,ident+1]=string(strcompress(max(magpol),/remove_all),format='(F10.1)')
+							props[18,ident+1]=string(strcompress(min(magpol),/remove_all),format='(F10.1)')
+							tbpos=string(strcompress(total(npix[where(magpol gt 0)]),/remove_all),format='(e11.1)')
+							props[19,ident+1]=strmid(tbpos,0,8)+strmid(tbpos,9,2)
+							tbneg=string('-'+strcompress(total(npix[where(magpol lt 0)]),/remove_all),format='(e10.1)')
+							props[20,ident+1]=strmid(tbneg,0,7)+strmid(tbneg,8,2)
+							phi=string(strcompress(mB*trummar*1e+16,/remove_all),format='(e11.1)')
+							props[21,ident+1]=strmid(phi,0,8)+strmid(phi,9,2)
+							phipos=string(strcompress(mBpos*trummar*1e+16,/remove_all),format='(e11.1)')
+							props[22,ident+1]=strmid(phipos,0,8)+strmid(phipos,9,2)
+							phineg=string('-'+strcompress(abs(mBneg*trummar*1e+16),/remove_all),format='(e10.1)')
+							props[23,ident+1]=strmid(phineg,0,7)+strmid(phineg,8,2)
```

Comment:
Instead of using `props[i, x]`... which probably if I ask you what `props[17, x]` keeps you won't know, use structures or dictionaries, so you could have something like `props.maxpol` or similar.